src/rtp/ext.rs

```diff
@@ -729,45 +820,82 @@ mod test {
         assert_eq!(ev.play_delay_min, ev2.play_delay_min);
         assert_eq!(ev.play_delay_max, ev2.play_delay_max);
     }
-}
 
-impl fmt::Debug for ExtensionMap {
-    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        write!(f, "Extensions(")?;
-        let joined = self
-            .0
-            .iter()
-            .enumerate()
-            .filter_map(|(i, v)| v.map(|v| (i + 1, v)))
-            .map(|(i, v)| format!("{i}={v}"))
-            .collect::<Vec<_>>()
-            .join(", ");
-        write!(f, "{joined}")?;
-        write!(f, ")")?;
-        Ok(())
+    #[test]
+    fn narrow_exts_audio() {
+        use Extension::*;
+
+        let mut e1 = ExtensionMap::standard();
+        let mut e2 = ExtensionMap::empty();
+        e2.set(14, TransportSequenceNumber);
+
+        println!("{:?}", e1.iter(false).collect::<Vec<_>>());
+
+        let correct = vec![(14, TransportSequenceNumber)];
+
+        e1.narrow(&e2.iter(true).collect::<Vec<_>>(), true);
+
+        assert_eq!(e1.iter(true).collect::<Vec<_>>(), correct);
+        assert_eq!(e2.iter(true).collect::<Vec<_>>(), correct);
+
+        // Ensure narrowing hasn't affected the video extensions
+        assert_eq!(
+            e1.iter(false).collect::<Vec<_>>(),
+            vec![
+                (2, AbsoluteSendTime),
+                (4, RtpMid),
+                (10, RtpStreamId),
+                (11, RepairedRtpStreamId),
+                (13, VideoOrientation),
+                (14, TransportSequenceNumber),
+            ]
+        );
     }
-}
 
-/// How the video is rotated.
-#[derive(Debug, Clone, Copy, PartialEq, Eq)]
-pub enum VideoOrientation {
-    /// Not rotated.
-    Deg0 = 0,
-    /// 90 degress clockwise.
-    Deg90 = 3,
-    /// Upside down.
-    Deg180 = 2,
-    /// 90 degrees counter clockwise.
-    Deg270 = 1,
-}
+    #[test]
+    fn narrow_exts_video() {
+        use Extension::*;
 
-impl From<u8> for VideoOrientation {
-    fn from(value: u8) -> Self {
-        match value {
-            1 => Self::Deg270,
-            2 => Self::Deg180,
-            3 => Self::Deg90,
-            _ => Self::Deg0,
-        }
+        let mut e1 = ExtensionMap::empty();
+        e1.set(3, TransportSequenceNumber);
+        e1.set(4, VideoOrientation);
+        e1.set(5, VideoContentType);
+        let mut e2 = ExtensionMap::empty();
+        e2.set(14, TransportSequenceNumber);
+        e2.set(12, VideoOrientation);
+
+        e1.narrow(&e2.iter(false).collect::<Vec<_>>(), false);
+
+        assert_eq!(
+            e1.iter(false).collect::<Vec<_>>(),
+            vec![(12, VideoOrientation), (14, TransportSequenceNumber)]
+        );
+        assert_eq!(
+            e2.iter(false).collect::<Vec<_>>(),
+            vec![(12, VideoOrientation), (14, TransportSequenceNumber)]
```

Comment:
This change wouldn't be okay if the remote had already agreed to have `3` mean `TransportSequenceNumber` I think.

It's definitely not okay for one id to map to different extensions and I think it's not okay to map several ids to the same extension

Reply:
Yeah. None of that would be ok. We're currently a bit better in PT negotiation with the `locked` property. Maybe we should do that here too?

Reply:
Yeah. None of that would be ok. We're currently a bit better in PT negotiation with the locked property. Maybe we should do that here too?

Reply:
Yes, that's why I had the proposed/negotiated distinction in webrtc-rs I think

Reply:
Yeah, I don't think we need to go that far. It's enough to just lock it ones we got the remote side confirmed.